component/task.go

```diff
@@ -0,0 +1,7 @@
+package component
+
+type TaskLaunchInfo struct {
```

Comment:
Docs, can also just put this in `component.go` if we want. Our `component.go` is a bit messy, so another option is moving more of the task stuff into here instead.

Reply:
Ah yes! Forgot to do a docs pass over here. On it.